### PR TITLE
add 112. Path Sum

### PR DESCRIPTION
**Pull request by tkkawa**
Please check my code.

### CODE REVIEW

File: 112_path-sum.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+class Solution:
+
+    def hasPathSum(self, root: TreeNode, targetSum: int) -> bool:
+        if not root:
+            return False
+
+        stack = deque([[root, targetSum - root.val]])
+        while stack:
+            cur_root, targetSum_rest = stack.pop()
```
Comment (ryuji0123) [inline 112_path-sum.py:26]: cur_root -> cur_node

**Review by ryuji0123 — CHANGES_REQUESTED** [review]

**Comment by tkkawa** [comment]
I modified the code.

**Review by ryuji0123 — CHANGES_REQUESTED** [review]
check cur_node only

**Comment by tkkawa** [comment]
I used the recursive function.
Please check my code.

File: 112_path-sum.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+class Solution:
+
+    def hasPathSum(self, root: TreeNode, targetSum: int) -> bool:
+        if root is None:
+            return False
+
+        elif root.left is None and root.right is None and root.val == targetSum:
+            return True
```
Comment (kobakobashu) [inline 112_path-sum.py:22]:
```
if not root.left and not root.right:
    return root.val == targetSum
```
I think this is better.

**Review by kobakobashu — CHANGES_REQUESTED** [review]

File: 112_path-sum.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+        else:
+            return self.hasPathSum(root.left, targetSum - root.val)\
+                or self.hasPathSum(root.right, targetSum - root.val)
```
Comment (kobakobashu) [inline 112_path-sum.py:26]: unnecessary else, I think.

**Comment by tkkawa** [comment]
I modified the code.
Please check my code.

**Review by kobakobashu — APPROVED** [review]
LGTM

**Pull request closed without merging**